core/src/typ.rs

```diff
@@ -1045,7 +1045,7 @@ impl Type {
             | TypeF::Var(_)
             | TypeF::Record(_)
             | TypeF::Enum(_) => true,
-            TypeF::Flat(rt) if matches!(*rt.term, Term::Var(_)) => true,
+            TypeF::Flat(rt) if rt.as_ref().is_atom() => true,
```

Comment:
This whole thing is a big `matches!()`
```
matches!(
    &self.typ,
    TypeF::Dyn
    | TypeF::Number
    | TypeF::Bool
    | TypeF::String
    | TypeF::Var(_)
    | TypeF::Record(_)
    | TypeF::Enum(_)
    | TypeF::Flat(rt) if rt.as_ref.is_atom()
)
```

Reply:
It's actually not, sneakily. Your suggested pattern isn't valid, because `rt` isn't bound in every branch :disappointed: